app/controllers/bookings_controller.rb

```diff
@@ -1,74 +1,60 @@
 class BookingsController < ApplicationController
-  before_action :set_booking, only: [:show, :edit, :update, :destroy]
 
-  # GET /bookings
-  # GET /bookings.json
   def index
-    @bookings = Booking.all
+    if current_admin.nil?
+      @bookings = Booking.includes(:city, :cleaner, :customer).where(customer_id: session[:customer_id])
+    else
+      redirect_to admins_path
+    end
   end
 
-  # GET /bookings/1
-  # GET /bookings/1.json
   def show
   end
 
-  # GET /bookings/new
   def new
-    @booking = Booking.new
+    if current_admin.nil?
+      @booking = Booking.new
+      @cities = City.all
+    else
+      redirect_to admins_path
+    end
   end
 
-  # GET /bookings/1/edit
   def edit
+    @city_cleaner = CitiesCleaner.where(city_id: params[:selectedCity]).pluck(:cleaner_id)
+    @cleaners = Cleaner.find(@city_cleaner)
+    render layout: false
   end
 
-  # POST /bookings
-  # POST /bookings.json
   def create
-    @booking = Booking.new(booking_params)
-
-    respond_to do |format|
+    @date = params[:booking].values[0] + "/" + params[:booking].values[1] + "/" + params[:booking].values[2]
+    @time = params[:booking].values[3] + ":" + params[:booking].values[4]
+    @date_time = (@date + " " + @time).to_datetime
+    @city_id = params[:city][:booking]
+    @cleaner_id = params[:city][:cleaners]
+    @records = Booking.where(datetime: @date_time - 2.hours..@date_time + 2.hours, cleaner_id: @cleaner_id)
+    if @records.count == 0
+      @booking = Booking.new(cleaner_id: @cleaner_id,city_id: @city_id,customer_id: session[:customer_id],datetime: @date_time)
       if @booking.save
-        format.html { redirect_to @booking, notice: 'Booking was successfully created.' }
-        format.json { render :show, status: :created, location: @booking }
+        CleanerMailer.booked().deliver
+        redirect_to url_for(:controller => :customers, :action => :index)
       else
-        format.html { render :new }
-        format.json { render json: @booking.errors, status: :unprocessable_entity }
+        @cities = City.all
```

Comment:
@cities = City.all

Can we move it to before filter as you have used it in different actions?